src/scripts/core/models/score-display.ts

```diff
@@ -0,0 +1,93 @@
+import { gsap } from "gsap";
+import { Container, Graphics, Text as PixiText } from "pixi.js";
+import { Token } from "./token";
+import { eventEmitter } from "../../../event-emitter";
+import { ScoreMaths } from "../services/score-maths";
+
+export class ScoreDisplay extends Container {
+
+    private score: number = 0;
+    private scoreText: PixiText = new PixiText(`${this.score}`);
+    private textContainer: Container = new Container();
+    private trackedHeight = -1;
+
+    constructor() {
+        super()
+
+        eventEmitter.on('onMatch', this.recordMatchedTokens, this)
+
+        const title: PixiText = new PixiText("SCORE");
+        this.scoreText.style = {
+            fill: "white"
+        };
+        const scoreCard = new Graphics();
+        scoreCard.beginFill(0xFFFFFF);
+        scoreCard.drawRoundedRect(0, 0, title.width, title.height, 3);
+        scoreCard.endFill();
+        scoreCard.addChild(title);
+        this.scoreText.position = {
+            x: scoreCard.width * 0.5 - this.scoreText.width,
+            y: scoreCard.height * 1.5
+        };
+        this.textContainer.addChild(scoreCard);
+        this.textContainer.addChild(this.scoreText);
+        this.addChild(this.textContainer);
+
+    }
+
+    private updateScore(targetScore: number) {
+        gsap.to(this, {
+            score: this.score + targetScore,
+            duration: 3,
+            onUpdate: this.showScore.bind(this),
```

Comment:
I run a function onUpdate that rounds the number displayed for the score in score_display
By default, gsap tweens the number and displays 6 decimal places through the transition and it looks really messy. Is there a better way to get an integer to display throughout the tween?

Reply:
I think what you have is a decent approach to your feature.
One optimisation I would reccommend is only assigning to `.text` if the number has actually changed after rounding.

```ts
private showScore(): void {
    if (!this._scoreText) { return; }

    const score = Math.round(this._score);
    if (this._scoreText.text !== score) {
        this._scoreText.text = score;
    }
}
```

I don't know if pixi has any internal optimisations to avoid reassigning the same text, but they do advise not to reassign it every frame as it can be expensive to do.
[Can read through all their performance reccommendations here.](https://pixijs.io/guides/production/performance-tips.html)

Reply:
I've used Liam's suggested fix 👍